types/helper.go

```diff
@@ -143,16 +148,47 @@ func strToInt(str string) (int64, error) {
 	} else if str[i] == '+' {
 		i++
 	}
-	r := int64(0)
+
+	var (
+		err    error
+		hasNum = false
+	)
+	r := uint64(0)
 	for ; i < len(str); i++ {
 		if !unicode.IsDigit(rune(str[i])) {
+			err = ErrTruncated
+			break
+		}
+		hasNum = true
+		if r >= uintCutOff {
+			r = 0
+			err = ErrBadNumber
```

Comment:
errors.Trace(err)

Reply:
If there are multiple error, say ErrBadNumber, in one function, we can't trace which is the actually cause of the error.
Do you remember I've ever write a error "invalid time format" without detail stack trace? And fix that introduce another PR, which could have been avoid. @coocood